packages/website/ts/pages/documentation/doc_page.tsx

```diff
@@ -128,7 +135,22 @@ export class DocPage extends React.Component<DocPageProps, DocPageState> {
 
         const versionFilePathToFetch = versionToFilePath[versionToFetch];
         const versionDocObj = await docUtils.getJSONDocFileAsync(versionFilePathToFetch, docBucketRoot);
-        const docAgnosticFormat = this.props.docsInfo.convertToDocAgnosticFormat(versionDocObj);
+        let docAgnosticFormat;
+        if (this.props.docsInfo.type === SupportedDocJson.TypeDoc) {
+            docAgnosticFormat = new TypeDocUtils(
+                versionDocObj as GeneratedDocJson,
+                this.props.docsInfo,
+            ).convertToDocAgnosticFormat();
+        } else if (this.props.docsInfo.type === SupportedDocJson.Solidity) {
+            // documenting solidity.
+            docAgnosticFormat = versionDocObj as DocAgnosticFormat;
+            // need to modify docsInfo like convertToDocAgnosticFormat() would do
```

Comment:
Lets prefix this comment with `HACK:`. This needs to be addressed in a future refactor.